text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1251 @@
+
+### Retain-Remove-Import Migration
+
+This strategy is employed when significant changes occur in the underlying resource, such as transitioning from
+a custom resource to a native L1 resource or resource property (e.g., Table to TableV2 migration). The CDK construct upgrade
+framework handles this process through several automated steps:
+
+- Validation of `RemovalPolicy.RETAIN` on the original construct to prevent accidental resource deletion.
+- Execution of the `cdk construct-upgrade` command, which performs necessary validations.
+- If validations pass, automatic triggering of deployment using CDK's import feature to link existing resources
+to the new construct definition.
+
+This approach preserves the state and data of existing resources while updating the infrastructure to use the
+modernized construct.
+
+## Construct Upgrade Validations
+
+Validations will ensure that user’s existing CDK stack and environment area ready for migration, flagging
+blockers like CloudFormation drift. Additionally, it validates that the migrated code is safe to deploy without
+causing downtime or replacement through CloudFromation template comparison and change set analysis. Here is a
+complete list of validations that CDK performs.
+
+### Drift Detection
+
+One of the validation workflow is to trigger CloudFormation drift detection for the CDK stacks. This feature
+is generic and can be applied to all CDK users. It will be an added functionality in the `cdk diff` command
+through the option `cdk diff --detect-drift`. The command will return the drift detection results in a format
+similar to and along with the current `cdk diff` output.
+
+> Note that this command will not fail if drift is detected. This is intentional, as `cdk diff` is meant to
+> provide information about changes and is designed to be status-agnostic.
+
+Sample output of `cdk diff --detect-drift` command is as follows:
+
+```sh
+$ cdk diff --detect-drift
+
+Drifts
+[~] AWS::Lambda::Function MyFunction drifted
+[~] AWS::DynamoDB::Table MyTable drifted
+
+Resources
+[-] AWS::DynamoDB::Table MyTable orphan
+[+] AWS::DynamoDB::GlobalTable MyTable import
+```
+
+The new `cdk construct-upgrade` CLI command will implicitly call the `cdk diff --detect-drift` command as part of
+its validation process. To avoid being overly restrictive, the validation will primarily focus on the main resources
+change. The main resources can be determined based on the `--target` option, more can be found in
+[Appendix C: Main Resources](#appendix-c-main-resources-computation).
+
+Drift detection is important to ensure construct upgrade safety by preventing accidental deployment issues when
+the CDK construct state does not match the actual deployed resource state. Without detecting drift, upgrades may
+unintentionally remove manual configurations made through the AWS Console, potentially triggering resource
+replacements, data loss, or availability downtime.
```

Comment:
Note that the only way to "resolve" drift manually is to manually undo those manual configurations via the AWS Console.

Changing your template to match reality may or may not do the right thing (it may even fail to deploy), and so it's unreliable and we cannot recommend it.